### DISCUSSION TITLE
diesel connection pool disconnected after a while

### DISCUSSION THREAD

**Discussion by jutbin** [diesel-rs/diesel#4165] [Q&A]
If the database pool connected by diesel is not used for a period of time, an error DatabaseError(Unknown, "Lost connection to MySQL server during query") will be reported.

**Comment by weiznich** [comment]
It's impossible to help you with a question like that if you don't provide a lot of details. Try providing a minimal reproducible example, otherwise there is no way to even know what you are trying to do.

**Reply by weiznich** [reply]
How can I use the provided example to reproduce your original error? It doesn't use the pool anywhere. 

Please consider to really always add all necessary information without having to be asked for them several times. If you want our help you need to make it as easy as possible for us to provide help. Given that this does not seem to be the case I will likely just stop responding to questions from you that miss the necessary details to be answerable. 

A minimal reproducible example should contain at least the following information:
* The relevant code (as minimal as possible, strip out any unneeded dependency like axum or tokio!).
* The complete untruncated error message as emitted by compiler (compile time error) or your application (runtime error)
* Information about your environment (database, operating system, compiler, etc)
* A detailed sequence of commands required to reproduce your problem, e.g run `cargo build` + `cargo run` + `wget foo`

(None of these points above is optional)

Ideally you provide something that anyone that might be able to answer the question is able to just `git clone && cargo check/run` on to reproduce exactly the same error as given in your description.

**Reply by jutbin** [reply]
Why does the connection pool set a minimum number of connections? Sometimes when I query the number of database connections, the minimum number of connections in the connection pool I set becomes 0.

```
    let pool = Pool::builder()
    .min_idle(Some(3))
    .max_size(15)
    .build(manager)
    .expect("Failed to create database pool");
```

**Reply by weiznich** [reply]
Given that the pool is not provided by diesel, but by the r2d2 crate that's something you need to ask the author of that crate.

**Reply by jutbin** [reply]
how do I make sure there's always at least one connection at any given instant

I use diesel::r2d2::Pool which is the features r2d2 of diesel

**Reply by weiznich** [reply]
`diesel::r2d2::Pool` is just a reexport of `r2d2::Pool` (as stated in the documentation), so again: Please use the support channels of the `r2d2` crate for questions about that type.

EDIT: That was my last answer on this topic, as you do not seem to be interested in either providing all relevant information as requested nor in taking advice.
